### PR TITLE
Add Unity contract interaction docs

### PR DESCRIPTION
**Pull request by alexandratran**
Add Unity smart contract interaction docs and make other minor improvements to the Unity docs, e.g. moving Unity SDK API into its own reference page.

See the following previews:
- [Unity SDK intro / how to set up](https://docs.metamask.io/contract-interaction/wallet/how-to/connect/set-up-sdk/gaming/unity/)
- [Unity smart contract interaction guide](https://docs.metamask.io/contract-interaction/wallet/how-to/interact-with-smart-contracts/unity/)
- [Unity SDK API reference](https://docs.metamask.io/contract-interaction/wallet/reference/sdk-unity-api/)

### CODE REVIEW

**Review by andreahaku — APPROVED** [review]
LGTM

File: wallet/how-to/connect/set-up-sdk/gaming/unity.md [hunk 1]
```diff
@@ -163,83 +188,6 @@ Create a new configuration by right-clicking on the project window and navigatin
...
-// Initialize using custom config, transport and socket provider
-var config = myMetaMaskConfig;
-var transport = new MyCustomTransport();
-var socketProvider = new MyCustomSocketProvider();
-MetaMaskUnity.Instance.Initialize(config, transport, socketProvider);
-```
-
-#### `SaveSession`
-
-This method saves the current session to the persistent storage.
```
Comment (mjsmike62) [inline wallet/how-to/connect/set-up-sdk/gaming/unity.md:203]: Maybe change  "...to the persistent storage" to "...to persistent storage"?

File: wallet/how-to/connect/set-up-sdk/gaming/unity.md [hunk 2]
```diff
@@ -163,83 +188,6 @@ Create a new configuration by right-clicking on the project window and navigatin
...
-This is useful when you want to save the session and restore it later.
-This is automatically called when the application quits, but you can also manually call it.
```
Comment (mjsmike62) [inline wallet/how-to/connect/set-up-sdk/gaming/unity.md:205]: Maybe change "This is automatically..." to "It is automatically..."?

File: wallet/how-to/connect/set-up-sdk/gaming/unity.md [hunk 3]
```diff
@@ -163,83 +188,6 @@ Create a new configuration by right-clicking on the project window and navigatin
...
-var config = myMetaMaskConfig;
-var transport = new MyCustomTransport();
-var socketProvider = new MyCustomSocketProvider();
-MetaMaskUnity.Instance.Initialize(config, transport, socketProvider);
-```
-
-#### `SaveSession`
-
-This method saves the current session to the persistent storage.
-This is useful when you want to save the session and restore it later.
```
Comment (mjsmike62) [inline wallet/how-to/connect/set-up-sdk/gaming/unity.md:204]: Maybe change "This is useful..." to "It is useful..."?

File: wallet/how-to/connect/set-up-sdk/gaming/unity.md [hunk 4]
```diff
@@ -163,83 +188,6 @@ Create a new configuration by right-clicking on the project window and navigatin
...
-This is automatically called when the application quits, but you can also manually call it.
-
-#### `LoadSession`
-
-This method loads the session from the persistent storage.
-This is useful when you want to restore the session after the application quits.
-This is automatically called when the application starts, but you can also manually call it.
```
Comment (mjsmike62) [inline wallet/how-to/connect/set-up-sdk/gaming/unity.md:211]: Maybe... "When the application starts, it automatically calls this method. You can also manually call this method."

Hmm... A little wordy to get rid of the passive voice.

File: wallet/how-to/connect/set-up-sdk/gaming/unity.md [hunk 5]
```diff
@@ -163,83 +188,6 @@ Create a new configuration by right-clicking on the project window and navigatin
...
-
-### MetaMaskWallet
-
-#### `Connect`
-
-This method connects to the MetaMask app.
```
Comment (mjsmike62) [inline wallet/how-to/connect/set-up-sdk/gaming/unity.md:217]: I saw earlier that "application" was used. I like "app". Our audience should understand "app".

File: wallet/how-to/connect/set-up-sdk/gaming/unity.md [hunk 6]
```diff
@@ -163,83 +188,6 @@ Create a new configuration by right-clicking on the project window and navigatin
...
-It renders a generated QR code in the user interface for your users to scan with MetaMask Mobile.
-After the user scans this QR code, a connect screen appears in MetaMask Mobile where the user can
-approve the connection with your game application.
```
Comment (mjsmike62) [inline wallet/how-to/connect/set-up-sdk/gaming/unity.md:220]: Formatting?

File: wallet/how-to/connect/set-up-sdk/gaming/unity.md [hunk 7]
```diff
@@ -163,83 +188,6 @@ Create a new configuration by right-clicking on the project window and navigatin
...
-
-#### `Disconnect`
-
-This method disconnects the user that is connected from the MetaMask app session.
```
Comment (mjsmike62) [inline wallet/how-to/connect/set-up-sdk/gaming/unity.md:224]: Maybe... "This method disconnects the user from the MetaMask session."

File: wallet/how-to/interact-with-smart-contracts/unity/contract-factory.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+## Set the contract factory
+
+To set the current contract factory, you can use `Contract.ContractFactory`:
+
+```csharp
+Contract.ContractFactory = new BackedTypeContractFactory();
+```
+
+Usually, you will never need to do this, unless you are creating a custom contract factory.
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/contract-factory.md:87]: Maybe... It's unlikely you'll need to do this, ...

File: wallet/how-to/interact-with-smart-contracts/unity/contract-factory.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+You should use the default contract factories for each runtime:
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/contract-factory.md:88]: Is it too strong to juse say... "Use the default..."

File: wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
+---
+sidebar_position: 1
+---
+
+# Contract interface
+
+When [interacting with smart contracts in Unity](index.md), the `IContract` interface defines an
+interface of a given contract.
+This interface includes all the functions of the contract, and optionally includes a constructor
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md:9]: How about this... "This interface includes all the functions of the contract, and includes an optional constructor..."?

File: wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+function and `Bytecode` field.
+
+You can use the [contract code generator](index.md#generate-contract-code) to generate a contract
+interface given a contract ABI or Hardhat artifact JSON file.
+
+To manually create a new contract interface, you must define a new interface that inherits from `IContract`.
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md:15]: Eliminate "you must"?

File: wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+You may optionally declare the [`BackedType` attribute](contract-factory.md#backed-type-contract-factory).
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md:16]: Replace "may" with "can"? For normalization, structure, and minimalization purposes, you may just want "Optionally, declare the..."

File: wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+	
+	[EvmMethodInfo(Name = "decimals", View = true)]  
+	[return: EvmParameterInfo(Type = "uint8")]  
+	Task<BigInteger> Decimals();
+}
+```
+
+### Use the Task return type
+
+It's recommend to always use `Task` as the return type when declaring contract functions, even if
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md:92]: Some say... "We recommend..." It's more active. 

If you keep it your way, then... recommend = recommended

File: wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+that `Task` completes, which may lead to a deadlock.
+
+### Define a contract constructor
+
+To define a constructor function, use the `EvmConstructorMethod` attribute at the top of the
+function declaring the constructor.
+The return type of the function must be the type of the interface, since a new instance of the
+interface is returned by the `Contract` class.
+
+You must also do one of the following:
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md:106]: Instead, say... "Also, do one of the following:" Cut down on words?

File: wallet/how-to/interact-with-smart-contracts/unity/index.md [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+  JSON file
+
+## Generate contract code
+
+You can use the MetaMask Unity contract code generator to generate
+[contract interface](contract-interface.md) files.
+The code generator automatically generates backed types for all contract interfaces to be used by
+the [contract factory](contract-factory.md#backed-type-contract-factory).
+
+In your Unity editor, go to **Tools > MetaMask > Contract ABI Converter**.
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/index.md:23]: go to = click ?

File: wallet/how-to/interact-with-smart-contracts/unity/index.md [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+This opens a new dialog box:
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/index.md:24]: Instead... "A new dialog box opens." ?

File: wallet/how-to/interact-with-smart-contracts/unity/index.md [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+switch contract addresses when the network changes.
+
+You can create a new template by going to **Assets > Create > MetaMask > Contract Templates** and
+selecting one of the following:
+
+- **ERC20**
+- **ERC721**
+- **ERC1155**
+
+Before using the contract template, you must configure the contract address to use for each chain:
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/index.md:97]: Maybe it's our style, but I would avoid from using "you must".

File: wallet/how-to/interact-with-smart-contracts/unity/index.md [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+
+</p>
+
+Once configured, you can use this scriptable object as a variable inside your scripts.
+The contract template has the same interface and usage as if you used `Contract.Attach`, but the
+setup and network switching is done for you.
+
+## Advanced topics
+
+To learn more about and customize your contract interactions using the Unity SDK, see the following topics:
```
Comment (mjsmike62) [inline wallet/how-to/interact-with-smart-contracts/unity/index.md:121]: You can just say... "see the following:"

File: wallet/reference/sdk-unity-api.md [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+
+This method loads the session from the persistent storage.
+This is useful when you want to restore the session after the application quits.
+This is automatically called when the application starts, but you can also manually call it.
+
+## MetaMaskWallet
+
+### Connect
+
+This method connects to the MetaMask app.
```
Comment (mjsmike62) [inline wallet/reference/sdk-unity-api.md:57]: Maybe just say "MetaMask" instead of "the MetaMask app"

File: wallet/reference/sdk-unity-api.md [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+It renders a generated QR code in the user interface for your users to scan with MetaMask Mobile.
+After the user scans this QR code, a connect screen appears in MetaMask Mobile where the user can
+approve the connection with your game application.
+
+### Disconnect
+
+This method disconnects the user that is connected from the MetaMask app session.
```
Comment (mjsmike62) [inline wallet/reference/sdk-unity-api.md:64]: Is it too simple to say... "This method disconnects the connected user from the MetaMask session."?

**Review by mjsmike62 — APPROVED** [review]
I am satisfied with the updated docs as per my edits/suggestions.

**Pull request merged**